LiveLines.Streaks/StreakService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using LiveLines.Api.Lines;
+using LiveLines.Api.Streaks;
+using LiveLines.Api.Users;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace LiveLines.Streaks;
+
+public class StreakService : IStreakService
+{
+    private readonly ILinesService _linesService;
+    private readonly IMemoryCache _expiringStreakCache;
+
+    private const string StreakCachePrefix = "Streak_";
+
+    public StreakService(ILinesService linesService, IMemoryCache expiringStreakCache)
+    {
+        _linesService = linesService;
+        _expiringStreakCache = expiringStreakCache;
+    }
+
+    private string GetStreakCacheKey(LoggedInUser user) => StreakCachePrefix + user;
+
+    public async Task<int> IncrementStreak(LoggedInUser user)
+    {
+        var newStreak = await GetStreak(user) + 1;
+        return _expiringStreakCache.Set(GetStreakCacheKey(user), newStreak);
```

Comment:
Concurrency

Reply:
Won't worry about this for now, since they can only post a line a day